panic in scalesets.go: boot diagnostics

/kind bug

[Before submitting an issue, have you checked the [Troubleshooting Guide](https://capz.sigs.k8s.io/topics/troubleshooting.html)?]

**What steps did you take and what happened:**
I upgraded from a previous version to 1.7.2. capz-controller-manager crashes because of a nil pointer.

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
       panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1a61c32]

goroutine 758 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
       /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:118 +0x1f4
panic({0x2001e60, 0x3947280})
       /usr/local/go/src/runtime/panic.go:884 +0x212
sigs.k8s.io/cluster-api-provider-azure/azure/services/scalesets.(*Service).validateSpec(0xc001867380, {0x26a7ed8?, 0xc001867500?})
       /workspace/azure/services/scalesets/scalesets.go:393 +0xab2
sigs.k8s.io/cluster-api-provider-azure/azure/services/scalesets.(*Service).Reconcile(0xc001867380, {0x26a7ed8?, 0xc001867470?})
       /workspace/azure/services/scalesets/scalesets.go:85 +0xcb
sigs.k8s.io/cluster-api-provider-azure/exp/controllers.(*azureMachinePoolService).Reconcile(0xc0018673e0, {0x26a7ed8?, 0xc0014f6180?})
       /workspace/exp/controllers/azuremachinepool_reconciler.go:68 +0x1b5
sigs.k8s.io/cluster-api-provider-azure/exp/controllers.(*AzureMachinePoolReconciler).reconcileNormal(0xc000784dc0, {0x26a7ea0?, 0xc0014bfb00?}, 0xc001ed75e0, 0xc000a77880)
       /workspace/exp/controllers/azuremachinepool_controller.go:290 +0x25e
sigs.k8s.io/cluster-api-provider-azure/exp/controllers.(*AzureMachinePoolReconciler).Reconcile(0xc000784dc0, {0x26a7ed8?, 0xc001a6b980?}, {{{0xc000896100?, 0xe?}, {0xc0008960f0?, 0x7?}}})
       /workspace/exp/controllers/azuremachinepool_controller.go:253 +0xd66
sigs.k8s.io/cluster-api-provider-azure/pkg/coalescing.(*reconciler).Reconcile(0xc0004b3400, {0x26a7ed8?, 0xc001a6b8c0?}, {{{0xc000896100?, 0x10?}, {0xc0008960f0?, 0x40dc87?}}})
       /workspace/pkg/coalescing/reconciler.go:109 +0x3ee
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x26a7e30?, {0x26a7ed8?, 0xc001a6b8c0?}, {{{0xc000896100?, 0x22200e0?}, {0xc0008960f0?, 0x404554?}}})
       /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:121 +0xc8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0002c8820, {0x26a7e30, 0xc000a62100}, {0x20b6820?, 0xc000543460?})
       /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:320 +0x33c
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0002c8820, {0x26a7e30, 0xc000a62100})
       /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:273 +0x1d9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
       /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:234 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
       /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:230 +0x333
```


**What did you expect to happen:**
No panic.

**Anything else you would like to add:**
We have nothing configured for boot diagnostics. 

**Environment:**

- cluster-api-provider-azure version: 1.7.2
- Kubernetes version: (use `kubectl version`): 1.23
- OS (e.g. from `/etc/os-release`): linux

--- a/azure/services/scalesets/scalesets.go
+++ b/azure/services/scalesets/scalesets.go
@@ -401,7 +401,7 @@ func (s *Service) validateSpec(ctx context.Context) error {
 	}
 
 	// Validate DiagnosticProfile spec
-	if spec.DiagnosticsProfile.Boot != nil {
+	if spec.DiagnosticsProfile != nil && spec.DiagnosticsProfile.Boot != nil {
 		if spec.DiagnosticsProfile.Boot.StorageAccountType == infrav1.UserManagedDiagnosticsStorage {
 			if spec.DiagnosticsProfile.Boot.UserManaged == nil {
 				return azure.WithTerminalError(fmt.Errorf("userManaged must be specified when storageAccountType is '%s'", infrav1.UserManagedDiagnosticsStorage))

--- a/azure/services/scalesets/scalesets_test.go
+++ b/azure/services/scalesets/scalesets_test.go
@@ -800,6 +800,27 @@ func TestReconcileVMSS(t *testing.T) {
 				s.HasReplicasExternallyManaged(gomockinternal.AContext()).Return(false)
 			},
 		},
+		{
+			name:          "should not panic when DiagnosticsProfile is nil",
+			expectedError: "",
+			expect: func(g *WithT, s *mock_scalesets.MockScaleSetScopeMockRecorder, m *mock_scalesets.MockClientMockRecorder) {
+				spec := newDefaultVMSSSpec()
+				spec.DiagnosticsProfile = nil
+				s.ScaleSetSpec().Return(spec).AnyTimes()
+
+				vmss := newDefaultVMSS("VM_SIZE")
+				vmss.VirtualMachineScaleSetProperties.VirtualMachineProfile.DiagnosticsProfile = nil
+
+				instances := newDefaultInstances()
+
+				setupDefaultVMSSInProgressOperationDoneExpectations(s, m, vmss, instances)
+				s.DeleteLongRunningOperationState(spec.Name, serviceName, infrav1.PutFuture)
+				s.DeleteLongRunningOperationState(spec.Name, serviceName, infrav1.PatchFuture)
+				s.UpdatePutStatus(infrav1.BootstrapSucceededCondition, serviceName, nil)
+				s.Location().AnyTimes().Return("test-location")
+				s.HasReplicasExternallyManaged(gomockinternal.AContext()).Return(false)
+			},
+		},
 	}
 
 	for _, tc := range testcases {

